ledger/simulation/tracer.go

```diff
@@ -63,6 +62,17 @@ func (tracer *cursorEvalTracer) relativeGroupIndex() int {
 	return tracer.relativeCursor[top]
 }
 
+// unifiedGroupIndex is a convenience method that gives index inside a group transaction from different run-mode:
+//  1. if we are running over application mode, then we can get information from cursorEvalTracer's relativeGroupIndex.
+//  2. otherwise, we should be under logic-sig mode, then we have to rely on cx.GroupIndex, for evaluation
+//     and verification of logic-sig happens ahead of transaction group evaluation.
+func (tracer *cursorEvalTracer) unifiedGroupIndex(cx *logic.EvalContext) int {
+	if cx.RunMode() == logic.ModeApp {
+		return tracer.relativeGroupIndex()
```

Comment:
I think this might always be the same as `cx.GroupIndex()`. If I remember right, I believe I introduced this before we had a public `GroupIndex` method on `EvalContext`.

Unless you are aware of some difference between the two?

Reply:
ah I mistaken you. I understood what you meant, as you are suggesting using `cx.GroupIndex` always, which makes sense. I am changing now.

Reply:
You're right, there is a difference because `relativeGroupIndex()` won't work for logic sigs, but I believe `cx.GroupIndex()` should work for all cases. Do you agree?

Reply:
yes, see [use cx.GroupIndex](https://github.com/algorand/go-algorand/pull/5360/commits/d392f9956fc0e9db84cbfa0e1542d5e3d8891c4e)